### PR TITLE
update bun package version

### PR DESCRIPTION
**Pull request by cdmistman**
Why
===

bun 1.0 is coming out soon, we'll wanna drum up hype

What changed
============

- update bun module to 0.7.14
- change runner to attempt using `package.json` first

Test plan
=========

- install this bun module in a Bun template repl
- `bun --version` should print `0.7.14`
- run button still works
- change `package.json#main` to be `src/index.ts`
- run button fails

Rollout
=======

_Describe any procedures or requirements needed to roll this out safely (or check the box below)_

- [ ] This is fully backward and forward compatible

### CODE REVIEW

File: pkgs/modules/bun/default.nix
Comment (airportyh) [inline pkgs/modules/bun/default.nix:11]: Why not use the absolute path to the executable?

File: pkgs/package-json-runner/default.nix
Comment (airportyh) [inline pkgs/package-json-runner/default.nix:3]: does bun have an equivalent to `npm run <command>`?

File: pkgs/modules/bun/default.nix
Comment (airportyh) [inline pkgs/modules/bun/default.nix:36]: fileParam should probably be changed to `optionalFileParam` because the package-json-runner can run without a file parameter as well.

File: pkgs/modules/bun/default.nix
Comment (cdmistman) [reply pkgs/modules/bun/default.nix:11]: to get the nice name in the runner's derivation name 😅 wasn't sure if i'd be bringing it back or not - it's coming back in one of the next commits :)

File: pkgs/package-json-runner/default.nix
Comment (cdmistman) [reply pkgs/package-json-runner/default.nix:3]: yes! good point lmao why would i do this

Comment (cdmistman) [reply pkgs/package-json-runner/default.nix:3]: ohhhhhhhh actually i see what's going on here. ok i'm not gonna keep trying this package-json-runner abstraction, that can happen later

**Review by airportyh — APPROVED** [review]
Approve for now to unblock. But I think you should change fileParam -> optionalFileParam before merging.

**Comment by cdmistman** [comment]
> should change fileParam -> optionalFileParam before merging

`fileParam` should probably accept `true`, `false`, or `optional`

**Review by airportyh — APPROVED** [review]

**Comment by airportyh** [comment]
> > should change fileParam -> optionalFileParam before merging
> 
> `fileParam` should probably accept `true`, `false`, or `optional`

I didn't want to change the API which would cause every module to change.

**Pull request merged**
